apisix/discovery/kubernetes/init.lua

```diff
@@ -298,88 +302,217 @@ local function create_endpoint_lrucache(endpoint_key, endpoint_port)
+        local id = conf.id
+
+        if ctx[id] then
+            error("duplicate id value")
+        end
+
+        local endpoint_dict = ngx.shared["kubernetes-" .. id]
+        if not endpoint_dict then
+            error(string.format("failed to get lua_shared_dict: ngx.shared.kubernetes-%s, ", id) ..
+                    "please check your APISIX version")
+        end
+
+        local apiserver, err = get_apiserver(conf)
+        if err then
+            error(err)
             return
         end
 
-        local ok, status = pcall(endpoints_informer.list_watch, endpoints_informer, apiserver)
+        local default_weight = conf.default_weight
 
-        local retry_interval = 0
-        if not ok then
-            core.log.error("list_watch failed, kind: ", endpoints_informer.kind,
-                    ", reason: ", "RuntimeException", ", message : ", status)
-            retry_interval = 40
-        elseif not status then
-            retry_interval = 40
+        local endpoints_informer, err = informer_factory.new("", "v1", "Endpoints", "endpoints", "")
+        if err then
+            error(err)
+            return
         end
 
-        ngx.timer.at(retry_interval, timer_runner)
+        setup_namespace_selector(conf, endpoints_informer)
+        setup_label_selector(conf, endpoints_informer)
+
+        endpoints_informer.on_added = on_endpoint_modified
+        endpoints_informer.on_modified = on_endpoint_modified
+        endpoints_informer.on_deleted = on_endpoint_deleted
+        endpoints_informer.pre_list = pre_list
+        endpoints_informer.post_list = post_list
+
+        ctx[id] = setmetatable({
+            endpoint_dict = endpoint_dict,
+            apiserver = apiserver,
+            default_weight = default_weight
+        }, { __index = endpoints_informer })
     end
 
-    ngx.timer.at(0, timer_runner)
+    for id, item in pairs(ctx) do
+        start_fetch(item)
+    end
+end
+
+
+local function multiple_mode_nodes(service_name)
+    local pattern = "^(.*)/(.*/.*):(.*)$" -- id/namespace/name:port_name
+    local match = ngx.re.match(service_name, pattern, "jo")
+    if not match then
+        core.log.error("get unexpected upstream service_name:　", service_name)
+        return nil
+    end
+
+    local id = match[1]
+    local endpoint_dict = ctx[id]
+    if not endpoint_dict then
+        core.log.error("id not exist")
+        return nil
+    end
+
+    local endpoint_key = match[2]
+    local endpoint_port = match[3]
+    local endpoint_version = endpoint_dict:get_stale(endpoint_key .. "#version")
+    if not endpoint_version then
+        core.log.info("get empty endpoint version from discovery DICT ", endpoint_key)
+        return nil
+    end
+
+    return endpoint_lrucache(service_name, endpoint_version,
+            create_endpoint_lrucache, endpoint_dict, endpoint_key, endpoint_port)
+end
+
+
+function _M.init_worker()
+    local discovery_conf = local_conf.discovery.kubernetes
```

Comment:
we should update `config-default.yaml`, added `id` attribute

Reply:
sorry i didn't understand what you mean

Reply:
here: https://github.com/apache/apisix/blob/2a44ba61b25224f585dc67caed7131d248cf9792/conf/config-default.yaml#L307-L336

only single_mode_nodes configuration here

Reply:
Ok ,get your point .